Code/Framework/AzToolsFramework/AzToolsFramework/Viewport/ViewBookmarkLoader.cpp

```diff
@@ -0,0 +1,552 @@
+
+                        // Initialize default locations to 0. This is a temporary solution to match the 12 locations of the legacy system.
+                        // Once there is a UI for the view bookmarks these lines should be removed.
+                        for (int i = 0; i < s_numOfDefaultLocationsInLevel; i++)
+                        {
+                            AZStd::string finalPath =
+                                "/" + m_bookmarkfileName + "/LocalBookmarks/" + AZStd::to_string(m_localBookmarkCount++);
+                            registry->SetObject(finalPath, ViewBookmark());
+                        }
+
+                        LoadViewBookmarks();
+                        return true;
+                    }
+                    else
+                    {
+                        LoadViewBookmarks();
+
+                        for (int i = 0; i < m_localBookmarks.size(); i++)
+                        {
+                            AZStd::string finalPath = "/" + m_bookmarkfileName + "/LocalBookmarks/" + AZStd::to_string(i);
+                            registry->SetObject(finalPath, m_localBookmarks.at(i));
+                        }
+
+                        return true;
+                    }
+                }
+                else // if the field is empty we don't have a file linked to the prefab, so we create one and we save it in the
+                     // component
+                {
+                    m_bookmarkfileName = GenerateBookmarkFileName();
+
+                    // Initialize default locations to 0
+                    for (int i = 0; i < s_numOfDefaultLocationsInLevel; i++)
+                    {
+                        AZStd::string finalPath = "/" + m_bookmarkfileName + "/LocalBookmarks/" + AZStd::to_string(m_localBookmarkCount++);
+                        registry->SetObject(finalPath, ViewBookmark());
+                    }
+
+                    LoadViewBookmarks();
+                    bookmarkComponent->SetLocalBookmarksFileName(m_bookmarkfileName);
+                    return true;
+                }
+            }
+        }
+        AZ_Warning("ViewBookmarkLoader", false, "Couldn't find a LocalViewBookmarkComponent");
+
+        return false;
+    }
+
+    bool ViewBookmarkLoader::SaveLocalBookmarkAtIndex(const ViewBookmark& bookmark, int index)
+    {
+
+        if (index < 0 || index > m_localBookmarkCount)
+        {
+            return false;
+        }
+
+        LoadDefaultLocalViewBookmarks();
+
+
+        AZStd::string finalPath = "/" + m_bookmarkfileName + "/LocalBookmarks/" + AZStd::to_string(index);
+
+        bool success = false;
+        if (auto registry = AZ::SettingsRegistry::Get())
+        {
+            success = registry->SetObject(finalPath, bookmark);
+        }
+
+        // If we managed to add the bookmark
+        if (success)
+        {
+            SaveBookmarkSettingsFile();
+            LoadViewBookmarks();
+        }
```

Comment:
Is it worth to do a warning in the case it's not successful to warn the user?

Same questions in the functions below that follow the same format.

Reply:
Any comment on this @AMZN-Igarri ?

Reply:
Yes! I added them. They are probably not showing cause I changed the name of the file to `LocalViewBookmarkLoader`.
```cppif (!success)
        {
             AZ_Warning(
                 "LocalViewBookmarkLoader", false, "View Bookmark x=%.4f, y=%.4f, z=%.4f couldn't be saved", bookmark.m_position.GetX(),
                 bookmark.m_position.GetY(), bookmark.m_position.GetZ());
            return false; 
        }
```

Reply:
Cool! I missed it, sorry. Thanks!